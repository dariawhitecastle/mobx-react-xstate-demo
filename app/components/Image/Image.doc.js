import { describe, PropTypes } from 'react-desc'

import { genericProps } from '../../utils/propTypes'

export const doc = Image => {
  const DocumentedImage = describe(Image)
    .description('An image.')
    .usage(
      `import { Image } from 'grommet';
<Image/>`,
    )
    .intrinsicElement('img')

  DocumentedImage.propTypes = {
    ...genericProps,
    fit: PropTypes.oneOf(['cover', 'contain']).description('How the image fills its container.'),
    fallback: PropTypes.string.description(
      'Specifies the URL of the fallback image used when src is failing to load',
    ),
    opacity: PropTypes.oneOfType([
      PropTypes.oneOf(['weak', 'medium', 'strong']),
      PropTypes.string,
      PropTypes.bool,
    ]).description('Transparency of the image.'),
  }

  return DocumentedImage
}

export const themeDoc = {
  'global.opacity.medium': {
    description: 'The value used when opacity is set to true.',
    type: 'number',
    defaultValue: '0.4',
  },
  'image.extend': {
    description: 'Any additional style for the Image.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
}
