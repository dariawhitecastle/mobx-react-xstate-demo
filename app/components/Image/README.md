<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createImage
An image.

## Usage

```javascript
import { Image } from 'grommet';
<Image/>
```

## Properties

**a11yTitle**

Custom title to be used by screen readers.

```
string
```

**alignSelf**

How to align along the cross axis when contained in
      a Box or along the column axis when contained in a Grid.

```
start
center
end
stretch
```

**gridArea**

The name of the area to place
    this inside a parent Grid.

```
string
```

**margin**

The amount of margin around the component. An object can
      be specified to distinguish horizontal margin, vertical margin, and
      margin on a particular side.

```
none
xxsmall
xsmall
small
medium
large
xlarge
{
  bottom: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  horizontal: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  left: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  right: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  top: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  vertical: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string
}
string
```

**fit**

How the image fills its container.

```
cover
contain
```

**fallback**

Specifies the URL of the fallback image used when src is failing to load

```
string
```

**opacity**

Transparency of the image.

```
weak
medium
strong
string
boolean
```
  
## Intrinsic element

```
img
```
## Theme
  
**global.opacity.medium**

The value used when opacity is set to true. Expects `number`.

Defaults to

```
0.4
```

**image.extend**

Any additional style for the Image. Expects `string | (props) => {}`.

Defaults to

```
undefined
```
