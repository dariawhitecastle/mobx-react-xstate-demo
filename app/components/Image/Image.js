/* eslint-disable react/prop-types */
/**
 *
 * Image
 *
 */

import React from 'react'
import { Image } from 'grommet'
import { doc } from './Image.doc'
import helpers from '../../utils/helpers'

function createImage({ a11yTitle, alignSelf, fallback, fit, gridArea, margin, opacity, src }) {
  return (
    <Image
      a11yTitle={a11yTitle}
      alignSelf={alignSelf}
      fallback={fallback}
      fit={fit}
      gridArea={gridArea}
      margin={margin}
      opacity={opacity}
      src={src}
    />
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createImage,
})
