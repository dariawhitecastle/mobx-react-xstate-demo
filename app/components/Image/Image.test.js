/**
 *
 * Tests for Image
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Image from './Image'

const renderComponent = (props = {}) => render(<Image {...props} />)

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})
