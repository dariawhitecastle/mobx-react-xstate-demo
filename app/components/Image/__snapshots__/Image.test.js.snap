// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <img
    class=""
  />
</div>
`;
